### ISSUE TITLE
Difference in test accuracy during training and evaluation

### ISSUE THREAD

**Issue by EvelynChee** [microsoft/robustdg#30]
Hi,

I am having some trouble reproducing the test accuracy obtained during training and that using the test script based on the saved checkpoint. It seems that it is because during the evaluation within training, `model.eval()` was not called first? This would mean the batch norm stats are updated based on the validation and test data. 

For example, I tried running the Chest X-ray experiments using the hybrid algorithm and the `get_test_accuracy` called by the following does not invoke any `model.eval()`: 
https://github.com/microsoft/robustdg/blob/3eee1730ae9efaae2bd763f9835ac540b9bde02a/algorithms/hybrid.py#L217-L218

Whereas, in the test script, the model is set to evaluation state during the loading of the model as shown below. When I removed this line during testing, I could then reproduce the test accuracy shown during training.
https://github.com/microsoft/robustdg/blob/3eee1730ae9efaae2bd763f9835ac540b9bde02a/evaluation/base_eval.py#L141

Please do correct me if I misunderstood any part. Thanks.

**Comment by divyat09** [comment]
Thanks for pointing this out! Indeed, it is an issue, there should have been a call to model.eval() during the evaluation function used within training. I quickly checked and this does not lead to any major change on the Rot/Fashion MNIST and PACS dataset, but perhaps more so on the Chest X-Ray dataset (due to the architecture being DenseNet instead of ResNet). For the Chest X-Ray dataset, there is not a lot of change when evaluating on RSNA as the test domain. Did you obtain similar results?

Further, Figure 3 from our work on privacy connection with DG (https://arxiv.org/pdf/2110.03369.pdf) should highlight the results on Chest X-Ray using the test_script. Maybe you can compare the results against them and check if there are any differences?

**Comment by EvelynChee** [comment]
Thanks for the reply. Will try out your suggestions.
